src/games/strategy/triplea/ai/proAI/ProTechAI.java

```diff
@@ -429,7 +429,11 @@ private static float determineEnemyBlitzStrength(final Territory blitzHere, fina
             continue;
           }
           for (final Unit u : neighbor.getUnits()) {
-            if (unitCondition.match(u) && MoveValidator.hasEnoughMovement(u, dist)) {
+            Route r1 = new Route();
+            for(Route r : routes){
+              r1 = Route.join(r1, r);
+            }
+            if (unitCondition.match(u) && Matches.UnitHasEnoughMovementForRoute(r1).match(u)) {
```

Comment:
DRY (don't repeat yourself)  violation.
Add a `Matches.UnitHasEnoughMovementForRoute(Collection<Route> routes )` method that does the join for you.